src/test/java/uk/gov/pay/adminusers/resources/InviteResourceOtpTest.java

```diff
@@ -2,58 +2,218 @@
+import static org.apache.commons.lang3.RandomStringUtils.random;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.nullValue;
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+import static uk.gov.pay.adminusers.app.util.RandomIdGenerator.newId;
 
 public class InviteResourceOtpTest extends IntegrationTest {
 
     private String code;
 
+    private static final String OTP_KEY = newId();
+    private static final int PASSCODE = new GoogleAuthenticator().getTotpPassword(base32().encode(OTP_KEY.getBytes()));
+    private static final String EMAIL = "invited-" + random(5) + "@example.com";
+    private static final String TELEPHONE_NUMBER = "+447999999999";
+    private static final String PASSWORD = "a-secure-password";
+
     @Before
     public void givenAnExistingInvite() {
-        code = InviteDbFixture.inviteDbFixture(databaseHelper).expired().insertInvite();
+        code = InviteDbFixture.inviteDbFixture(databaseHelper)
+                .withEmail(EMAIL)
+                .withOtpKey(OTP_KEY)
+                .expired()
+                .insertInvite();
     }
 
     @Test
     public void generateOtp_shouldSucceed_evenWhenTokenIsExpired_sinceItShouldBeValidatedOnGetInvite() throws Exception {
 
-        String telephoneNumber = "+447999999999";
-
         ImmutableMap<Object, Object> invitationRequest = ImmutableMap.builder()
-                .put("telephone_number", telephoneNumber)
-                .put("password", "a-secure-password")
+                .put("code", code)
+                .put("telephone_number", TELEPHONE_NUMBER)
+                .put("password", PASSWORD)
                 .build();
 
         givenSetup()
                 .when()
                 .body(mapper.writeValueAsString(invitationRequest))
                 .contentType(ContentType.JSON)
-                .post(String.format(INVITES_OTP_RESOURCE_URL, code))
+                .post(INVITES_GENERATE_OTP_RESOURCE_URL)
                 .then()
                 .statusCode(OK.getStatusCode());
     }
 
     @Test
     public void generateOtp_shouldFail_whenInviteDoesNotExist() throws Exception {
 
-        String telephoneNumber = "+447999999999";
-
         ImmutableMap<Object, Object> invitationRequest = ImmutableMap.builder()
-                .put("telephone_number", telephoneNumber)
-                .put("password", "a-secure-password")
+                .put("code", "not-existing-code")
+                .put("telephone_number", TELEPHONE_NUMBER)
+                .put("password", PASSWORD)
                 .build();
 
         givenSetup()
                 .when()
                 .body(mapper.writeValueAsString(invitationRequest))
                 .contentType(ContentType.JSON)
-                .post(format(INVITES_OTP_RESOURCE_URL, "not-existing-code"))
+                .post(INVITES_GENERATE_OTP_RESOURCE_URL)
+                .then()
+                .statusCode(BAD_REQUEST.getStatusCode());
```

Comment:
I don't think these 2 tests are really necessary (just one is enough). As discussed